synapse/storage/client_ips.py

```diff
@@ -142,9 +150,8 @@ def get_last_seen(txn):
             last_seen = int(self.clock.time_msec()) * 2
         else:
             last = False
-            last_seen = last_seen[0]
 
-        def remove(txn, last_seen_progress, last_seen):
+        def remove(txn, begin_last_seen, end_last_seen):
```

Comment:
The fact that these params have the same names as their counterparts in the parent scope is a recipe for confusion imho.

Could you either:
* Remove the params and just reference the vars via the parent scope
* Rename the params so that they don't clash with the parent scope